decidim-comments/spec/forms/comment_form_spec.rb

```diff
@@ -103,6 +107,60 @@ module Comments
           end
         end
       end
+
+      shared_examples "allows commenting" do
+        it "allows commenting" do
+          expect(subject.send(:commentable_can_have_comments)).to be_nil
+          expect(subject).to be_valid
+        end
+      end
+
+      shared_examples "does not allow commenting" do
+        it "does not allow comments" do
+          expect(subject.send(:commentable_can_have_comments)).not_to be_nil
+          expect(subject).not_to be_valid
+        end
+      end
+
+      describe "#commentable_can_have_comments" do
+        let(:accepts_new_comments) { true }
+
+        before do
+          allow(commentable).to receive(:accepts_new_comments?).and_return(accepts_new_comments)
+        end
+
+        it_behaves_like "allows commenting"
+
+        context "when no comments are accepted" do
+          let!(:accepts_new_comments) { false }
+
+          it_behaves_like "does not allow commenting"
+
+          context "when user is admin" do
+            let(:user) { create(:user, :admin, :confirmed, organization:) }
+
+            it_behaves_like "allows commenting"
+          end
+
+          context "when user is user manager" do
+            let(:user) { create(:user, :user_manager, :confirmed, organization:) }
+
+            it_behaves_like "allows commenting"
+          end
+
+          context "when user is moderator in the same participatory space" do
+            let!(:moderator_role) { create(:assembly_user_role, user:, assembly:, role: :moderator) }
+
+            it_behaves_like "allows commenting"
+          end
+
+          context "when user is moderator in another participatory space" do
```

Comment:
This may be a happy path. Could we add some other specs like "what happens when the space is a process"?